tools/c7n_sphinxext/c7n_sphinxext/docgen.py

```diff
@@ -37,10 +38,21 @@ def get_environment():
     env.globals['ename'] = ElementSchema.name
     env.globals['edoc'] = ElementSchema.doc
     env.globals['eschema'] = CustodianSchema.render_schema
+    env.globals['eperm'] = eperm
     env.globals['render_resource'] = CustodianResource.render_resource
     return env
 
 
+def eperm(r, el):
+    if el.permissions:
+        return el.permissions
+    print('no permission defined %s %s' % (el.type, el))    
+    cfg = Config.empty(session_factory=lambda : 1)
+    rtype = r({}, Config.empty(options=cfg))
+    e = el({}, rtype)
+    return e.get_permissions()
```

Comment:
this only gets default permissions, in many cases actual permissions are dependent on configuration.. so this a starter only :-(